app/views/devise/registrations/edit.haml

```diff
@@ -177,10 +177,11 @@
                 .tab-pane.fade#premium_settings
                     %h3.strong Premium Settings
                     %hr
-                    %p
-                        Purchase a premium rank from our
-                        %a{:href => main_app.shop_path} shop page
-                        to have access to Premium Features.
+                    - if !@user.premium?
```

Comment:
In Ruby, you could also do `unless @user.premium?`

Reply:
Did it like this since its like that in the validation code